self-core-impl/src/main/java/com/selfxdsd/core/JsonResources.java

```diff
@@ -420,5 +439,24 @@ public JsonArray asJsonArray() {
                 new StringReader(this.body)
             ).readArray();
         }
+
+        @Override
+        public Map<String, List<String>> headers() {
+            return this.headers;
+        }
+
+        @Override
+        public Resource newInstance(
+            final int status,
+            final JsonValue body,
+            final Map<String, List<String>> headers
+        ) {
+            return new JsonResponse(status, body.toString(), headers);
+        }
```

Comment:
@criske Why is this method needed? :D

Reply:
@amihaiemil In CachingJsonResources, the NOT_MODIFIED status Resource response is empty. So we need to "modify" that Resource with an OK status and the cached body, while keeping its headers. Since Resource is an interface, a factory is needed.

```java
 resource.newInstance(
            HttpURLConnection.HTTP_OK,
            Json.createReader(new StringReader(body)).readObject(),
            resource.headers()
        );
```
But now that I'm think of it, a builder based on "this" resource will be more intuitive. The above code will look much better. 
```java
resource.newBuilder().status(HTTP_OK).body(cachedBody).build()
```
Plus I think there is a bug there too: it assumes that `body` is an object, but also it could be an array. It should be
` Json.createReader(new StringReader(body)).read()`.

Reply:
@criske newInstance is a fine name, leave it like that :D 

I hade words such as builder, service, instance etc. But newInstance is the least bad haha

Reply:
@amihaiemil sorry, I've seen this after my commit ... let me know what to revert/rename  etc :D